src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -180,11 +195,16 @@ AzureOAuthService getOAuthService() {
 
     private String getRootUrl() {
         Jenkins jenkins = Jenkins.getInstance();
+        if (isFromRequest()) {
+            return StringUtils.stripEnd(jenkins.getRootUrlFromRequest(), "/");
+        }
         return StringUtils.stripEnd(jenkins.getRootUrl(), "/");
     }
 
     @DataBoundConstructor
-    public AzureSecurityRealm(String tenant, String clientId, String clientSecret, int cacheDuration) {
+    public AzureSecurityRealm(String tenant, String clientId, String clientSecret, int cacheDuration,
+                              boolean fromRequest)
```

Comment:
There is no need to modify databound constructor also your breaking binary compatibility.

When adding new non mandatory fields I suggest using the `DataBoundSetter`
Please add `@DataBoundSetter` to `setFromRequest`